src/main/java/uk/ac/ebi/eva/pipeline/io/mappers/VariantVcfFactory.java

```diff
@@ -225,26 +225,35 @@ protected VariantKeyFields normalizeLeftAlign(String chromosome, int position, S
 
     protected void parseSplitSampleData(Variant variant, String fileId, String studyId, String[] fields,
                                         String[] alternateAlleles, String[] secondaryAlternates,
-                                        int alternateAlleleIdx) throws NonStandardCompliantSampleField {
+                                        int alternateAlleleIdx) throws NonStandardCompliantSampleField, NotAVariantException {
         String[] formatFields = variant.getSourceEntry(fileId, studyId).getFormat().split(":");
 
         for (int i = 9; i < fields.length; i++) {
             Map<String, String> map = new TreeMap<>();
-
+            int countReferenceGenotypes = 0;
             // Fill map of a sample
             String[] sampleFields = fields[i].split(":");
 
             // Samples may remove the trailing fields (only GT is mandatory),
             // so the loop iterates to sampleFields.length, not formatFields.length
             for (int j = 0; j < sampleFields.length; j++) {
+            	if(formatFields[0] == "GT"){
+            		if(sampleFields[j].equals("./.") || sampleFields.equals("0/0") || sampleFields.equals("0|0")){
+            			countReferenceGenotypes++;
+            		}
+            	}
                 String formatField = formatFields[j];
                 String sampleField = processSampleField(alternateAlleleIdx, formatField, sampleFields[j]);
 
                 map.put(formatField, sampleField);
-            }
+                if(countReferenceGenotypes == sampleFields.length){
```

Comment:
`sampleFields` is the array containing the fields associated with a single sample, not all of them.